src/main/scala/org/zalando/spark/jsonschema/SchemaConverter.scala

```diff
@@ -160,12 +160,39 @@ object SchemaConverter {
     schema.add(getJsonName(json).getOrElse(name), dataType, nullable = nullable)
   }
 
+  private def getArrayDataType(inputJson: JsObject, contentPath: JsPath): DataType = {
```

Comment:
This function duplicates the whole pattern matching from the 'addJsonField'-function. You could save a lot code by reusing the pattern matching there. So instead of defining this new function here, I would propose to incorporate your changes by modifying the 'addJsonField'-function. This may look like:

```
  private def addJsonField(schema: StructType, inputJson: JsObject, name: String): StructType = {
    val json = checkRefs(inputJson)
    val (dataType, nullable) = processType(json, name)

    schema.add(getJsonName(json).getOrElse(name), dataType, nullable = nullable)
  }

  private def processType(json: JsObject, name: String): (DataType, Boolean) = {

    val fieldType = getJsonType(json, name)
    TypeMap(fieldType.typeName) match {

      case dataType: DataType =>
        (dataType, fieldType.nullable)

      case ArrayType =>
        val innerJson = checkRefs((json \ SchemaArrayContents).as[JsObject])
        val dataType = ArrayType(processType(innerJson, "")._1)
        (dataType, getJsonType(json, name).nullable)

      case StructType =>
        val dataType = getDataType(json, JsPath \ SchemaStructContents)
        (dataType, getJsonType(json, name).nullable)
    }
  }
```